documentation/diagnostics-client-library-instructions.md

```diff
@@ -0,0 +1,522 @@
+public void WriteDump(DumpType dumpType, string dumpPath=null, bool logDumpGeneration=false);
+```
+
+Request a dump for post-mortem debugging of the target application. The type of the dump can be specified using the [`DumpType`](#enum-dumptype) enum.
+
+* `dumpType` : Type of the dump to be requested.
+
+* `dumpPath` : The path to the dump to be written out to.
+* `logDumpGeneration` : If set to `true`, the target application will write out diagnostic logs during dump generation.
+
+
+
+
+
+```csharp
+public void AttachProfiler(TimeSpan attachTimeout, Guid profilerGuid, string profilerPath, byte[] additionalData=null);
+```
+
+Request to attach an ICorProfiler to the target application. 
+
+* `attachTimeout` : A `TimeSpan` after which attach will be aborted.
+* `profilerGuid` :  `Guid` of the ICorProfiler to be attached.
+* `profilerPath `  : Path to the ICorProfiler dll to be attached.
+* `additionalData` : Optional additional data that can be passed to the runtime during profiler attach.
+
+
+
+```csharp
+public static IEnumerable<int> GetPublishedProcesses();
+```
+
+Get an `IEnumerable` of process IDs of all the active .NET processes that can be attached to.
+
+
+
+
+
+#### class EventPipeProvider
+
+```cs
+public class EventPipeProvider
+{
+    public EventPipeProvider(
+        string name,
+        EventLevel eventLevel,
+        long keywords = 0,
+        IDictionary<string, string> arguments = null)
+
+    public string Name { get; }
+
+    public EventLevel EventLevel { get; }
+
+    public long Keywords { get; }
+
+    public IDictionary<string, string> Arguments { get; }
+
+    public override string ToString();
+
+    public override bool Equals(object obj);
+
+    public override int GetHashCode();
+
+    public static bool operator ==(Provider left, Provider right);
+
+    public static bool operator !=(Provider left, Provider right);
+}
+```
+
+
+
+```csharp
+public EventPipeProvider(string name,
+                         EventLevel eventLevel,
+                         long keywords = 0,
+                         IDictionary<string, string> arguments = null)
+```
+
+Creates a new instance of `EventPipeProvider` with the given provider name, [EventLevel](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventlevel), keywords, and arguments. 
+
+#### Properties
+
+
+
+```csharp
+public string Name { get; }
+```
+
+The name of the Provider
+
+
+
+```csharp
+public EventLevel EventLevel { get; }
+```
+
+The EventLevel of the given instance of [`EventPipeProvider`](#class-eventpipeprovider).
+
+
+
+```csharp
+public long Keywords { get; }
+```
+
+A long that represents bitmask for keywords of the EventSource. 
+
+
+
+```csharp
+public IDictionary<string, string> Arguments { get; }
+```
+
+An `IDictionary` of key-value pair string representing optional arguments to be passed to EventSource representing the given `EventPipeProvider`.
+
+
+
+#### Methods
+
+```csharp
+public override string ToString();
```

Comment:
Usually ToString, Equals, GetHashCode, and operator overloads can be omitted from docs because there are well understood conventions about how these methods are going to behave. I would only explicitly document them if they had surprising behavior.